tests/queries/quote_rewards_test_db.sql

```diff
@@ -0,0 +1,67 @@
+DROP TABLE IF EXISTS orders;
+DROP TYPE IF EXISTS OrderKind;
+DROP TABLE IF EXISTS order_quotes;
+DROP TABLE IF EXISTS trades;
+
+
+-- orders table
+CREATE TYPE OrderKind AS ENUM ('buy', 'sell');
+
+CREATE TABLE IF NOT EXISTS orders
+(
+    uid bytea PRIMARY KEY,
+    sell_amount numeric(78,0) NOT NULL,
+    buy_amount numeric(78,0) NOT NULL,
+    fee_amount numeric(78,0) NOT NULL,
+    kind OrderKind NOT NULL,
+    partially_fillable boolean NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS order_quotes
+(
+  order_uid bytea PRIMARY KEY,
+  sell_amount numeric(78, 0) NOT NULL,
+  buy_amount numeric(78, 0) NOT NULL,
+  solver bytea NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS trades
+(
+  block_number bigint NOT NULL,
+  log_index bigint NOT NULL,
+  order_uid bytea NOT NULL,
+  sell_amount numeric(78, 0) NOT NULL,
+  buy_amount numeric(78, 0) NOT NULL,
+
+  PRIMARY KEY (block_number, log_index)
+);
+
+
+TRUNCATE orders;
+TRUNCATE order_quotes;
+TRUNCATE trades;
+
+
+INSERT INTO orders (uid, sell_amount, buy_amount, fee_amount, kind, partially_fillable)
+VALUES ('\x01'::bytea, 95000000, 94000000000000000000, 5000000, 'sell', 'f'), -- normal sell market order
+('\x02'::bytea, 101000000, 100000000000000000000, 5000000, 'buy', 'f'), -- normal buy market order
+('\x03'::bytea, 100000000, 100000000000000000000, 0, 'sell', 't'), -- partially fillable sell limit order
+('\x04'::bytea, 100000000, 100000000000000000000, 0, 'buy', 't'), -- partially fillable buy limit order
+('\x05'::bytea, 100000000, 94000000000000000000, 0, 'sell', 'f'), -- in market sell limit order
+('\x06'::bytea, 106000000, 100000000000000000000, 0, 'buy', 'f'); -- in market buy limit order
+
+INSERT INTO order_quotes (order_uid, sell_amount, buy_amount, solver)
```

Comment:
the order_quotes table does have a fee entry but i am fine with assuming there is none here.

Reply:
If I understand correctly, the `order_quotes` table does not have a fee entry. It does have `gas_amount`, `gas_price`, and `sellt_token_price` but due to hooks this is not enough to recover fees.

For market orders we can use the amount in the `orders` or `trades` table. For limit orders I am not sure what to do. It there a way to recover the estimated fee amount or does quoting maybe work differently for such orders?

Reply:
Oh you are right, it is implicit.